src/state.rs

```diff
@@ -237,6 +210,46 @@ impl State {
 
         x
     }
+
+    pub fn persistent_read_only<T, X>(
```

Comment:
nit: In Rust, `persistent` / `persistent_mut` is often used for read-only and read-write getters.  As this takes a callback instead of returning a reference, it could be called `with_persistent` / `with_persistent_mut`.

Reply:
That's a good point - naming _is_ important. Changed accordingly. 
I have added `try` prefix nevertheless for the `mut` case, as it seems coherent with other - does this make sense for you? Read only state is infallible for now.

Reply:
That’s fine.  As Rust forces us to handle returned `Result`s, adding `try` to the method name is not a requirement, but it can make sense to highlight that the operations is fallible, especially if there are fallible and infallible versions of the same operation.